apps/google-analytics-4/frontend/src/helpers/externalUrlHelpers/externalUrlHelpers.ts

```diff
@@ -43,3 +57,13 @@ export const encodeDataFiltersParameters = (str: string) => {
 
   return encodedStr;
 };
+
+const formatDateParams = (dateString: string) => {
+  // Dates need to be formatted from YYYY-MM-DD to YYYYMMDD and month must be zero-padded
```

Comment:
I tested this with single digit months and dates (e.g. 2023-1-3) and learned that the month needs to be zero-padded (e.g. 2023-01-3). For the date, it's ok that it's a single digit and it still works.